zio-http/src/main/scala/zhttp/http/URL.scala

```diff
@@ -24,6 +24,8 @@ final case class URL(
     case abs: URL.Location.Absolute => Option(abs.port)
   }
 
+  lazy val relative: URL = self.copy(kind = URL.Location.Relative)
```

Comment:
Can make this package-private. It's a bit too low-level API. What do you think?
Lazy vals are also a bit slower for first-time access. I would prefer a `def` and save it in a value in case of hot paths.